### PR TITLE
feat: [#1300] tests for jms subscriber

### LINKED ISSUE #1300: Write tests for jms subscriber

### PR DESCRIPTION
**Pull request by wiktorflorian**
Closes #1300 

Added complex tests for [app/services/jms/connector.py](https://raw.githubusercontent.com/cyfronet-fid/eosc-search-service/development/transform/app/services/jms/connector.py) and [app/transform/live_update/training_ig.py](https://raw.githubusercontent.com/cyfronet-fid/eosc-search-service/development/transform/app/transform/live_update/training_ig.py).

### CODE REVIEW

File: transform/tests/services/jms/conftest.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+
+@pytest.fixture
+def subscription_id() -> str:
+    """Generate a unique subscription ID."""
+    return f"{settings.STOMP_CLIENT_NAME}-{uuid.uuid4()}"
+
+
+@pytest.fixture
+def jms_handler(mock_stomp_connection: MagicMock) -> JMSMessageHandler:
```
Comment (Michal-Kolomanski) [inline transform/tests/services/jms/conftest.py:27]: `def jms_handler(mock_stomp_connection: MagicMock) -> JMSMessageHandler:` -> 
`def jms_handler(mock_stomp_connection: MagicMock, subscription_id: str) -> JMSMessageHandler:`
      `    subscription_id=subscription_id`

**Review by Michal-Kolomanski — COMMENTED** [review]
Impressive tests! 💪 although please fix tests that are failing

File: transform/tests/services/jms/conftest.py (hunk 1 above)
Comment (wiktorflorian) [reply transform/tests/services/jms/conftest.py]: Good catch, I forgot about it even after all that struggle with tests 👀

File: transform/app/services/jms/connector.py [hunk 2]
```diff
@@ -31,7 +31,7 @@ def subscription_condition(connection: stomp.Connection) -> bool:
     Returns:
         bool: True if the connection is active, False otherwise.
     """
-    return connection.is_connected()
+    return connection is not None and connection.is_connected()
```
Comment (Michal-Kolomanski) [inline transform/app/services/jms/connector.py:34]: `return connection and connection.is_connected()`

**Review by Michal-Kolomanski — APPROVED** [review]
👏

**Pull request merged**